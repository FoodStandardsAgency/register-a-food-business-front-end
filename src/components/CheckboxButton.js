import { FONT_SIZE, MEDIA_QUERIES } from "@govuk-react/constants";
import styled from "react-emotion";

const CheckboxButton = styled(buttom)({
  fontSize: FONT_SIZE.SIZE_16,
  [MEDIA_QUERIES.LARGESCREEN]: {
    fontSize: FONT_SIZE.SIZE_16
  }
});

export default CheckboxButton;
