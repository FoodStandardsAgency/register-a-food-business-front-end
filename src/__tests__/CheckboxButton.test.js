import CheckboxButton from "../components/CheckboxButton";
import { shallow } from "enzyme";

describe("<Checkbox />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<CheckboxButton />);
    expect(wrapper.length).toBe(1);
  });
});
