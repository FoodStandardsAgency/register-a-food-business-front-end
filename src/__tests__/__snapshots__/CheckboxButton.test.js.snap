// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> matches the previous snapshot 1`] = `
.emotion-6 {
  display: block;
  position: relative;
  padding: 0 0 0 38px;
  margin-bottom: 20px;
}

@media only screen and (min-width:641px) {
  .emotion-6 {
    margin-bottom: 30px;
  }
}

.emotion-0 {
  position: absolute;
  left: 0;
  top: 0;
  width: 38px;
  height: 38px;
  z-index: 1;
  margin: 0;
  zoom: 1;
  opacity: 0;
}

.emotion-0:focus + span:before {
  box-shadow: 0 0 0 4px #ffbf47;
}

.emotion-0:visited {
  background: #b10e1e;
  color: #b10e1e;
}

.emotion-3 {
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  text-transform: none;
  font-size: 16px;
  line-height: 1.25;
  cursor: pointer;
  padding: 8px 10px 9px 12px;
  display: block;
  color: #0b0c0c;
  margin-bottom: 20px;
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    font-size: 19px;
    line-height: 1.3;
  }
}

.emotion-3::before {
  content: '';
  display: block;
  border: 2px solid #0b0c0c;
  background: transparent;
  width: 34px;
  height: 34px;
  position: absolute;
  top: 0;
  left: 0;
}

.emotion-3::after {
  content: '';
  border: solid;
  border-width: 0 0 5px 5px;
  background: transparent;
  border-top-color: transparent;
  width: 17px;
  height: 7px;
  position: absolute;
  top: 10px;
  left: 8px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
  zoom: 1;
  opacity: 0;
}

.emotion-3.checked::after {
  opacity: 1;
}

@media only screen and (min-width:641px) {
  .emotion-3 {
    margin-bottom: 30px;
  }
}

<label
  className="emotion-2 emotion-6 emotion-7"
>
  <button
    className="emotion-0 emotion-1"
  />
  <span
    className="emotion-2 emotion-3 emotion-4"
  />
</label>
`;
